javagen/src/main/java/com/azure/autorest/util/ClientModelUtil.java

```diff
@@ -547,4 +549,33 @@ public static boolean isWireTypeMismatch(ClientModelProperty clientModelProperty
             }
         }
     }
+
+    /**
+     * Checks where {@code CoreToCodegenBridgeUtils} should be generated.
+     * <p>
+     * At this time it is required if {@link JavaSettings#isStreamStyleSerialization()} is true and the model uses
+     * either {@code ResponseError} or {@link Duration} as both types have special serialization requirements that
+     * aren't exposed by azure-core.
+     *
+     * @param model the model
+     * @param settings Java settings
+     * @return Whether to generate the {@code CoreToCodegenBridgeUtils} utility class.
+     */
+    public static boolean generateCoreToCodegenBridgeUtils(ClientModel model, JavaSettings settings) {
+        if (!settings.isStreamStyleSerialization()) {
+            return false;
+        }
+
+        boolean ret = model.getProperties().stream()
+            .anyMatch(p -> p.getClientType() == ClassType.ResponseError || p.getClientType() == ClassType.Duration);
+
+        // flatten properties
+        if (!ret && settings.getClientFlattenAnnotationTarget() == JavaSettings.ClientFlattenAnnotationTarget.NONE) {
+            ret = model.getPropertyReferences().stream()
+                .filter(ClientModelPropertyReference::isFromFlattenedProperty)
+                .anyMatch(p -> p.getClientType() == ClassType.ResponseError || p.getClientType() == ClassType.Duration);
```

Comment:
Is there possible other cases? E.g. `List<Duration>` or `Map<>`?

Reply:
Yeah, it could be used in `List` and `Map` but I didn't see any cases at this time.